Mathlib/Combinatorics/SimpleGraph/ConcreteColorings.lean

```diff
@@ -19,6 +19,16 @@ This file defines colorings for some common graphs
 
 namespace SimpleGraph
 
+theorem two_le_chromaticNumber_of_adj {α} {G : SimpleGraph α} {u v : α} (hAdj : G.Adj u v) :
+    2 ≤ G.chromaticNumber := by
+  refine le_of_not_lt ?_
+  intro h
+  have hc: G.Colorable 1 := chromaticNumber_le_iff_colorable.mp (Order.le_of_lt_add_one h)
+  let c : G.Coloring (Fin 1) := hc.some
```

Comment:
Since you spend quite a bit of time just fiddling with the logic here, this suggests it could be useful to have a lemma saying `n < G.chromaticNumber` if G is not n-colourable.

Reply:
For using it I ended up having to do all this:
```lean
  rw [show (2 : ℕ∞) = 1 + 1 by rfl]
  rw [ENat.add_one_le_iff ENat.one_ne_top]
  rw [show (1 : ℕ∞) = (1 : ℕ) by rfl]
  rw [le_chromaticNumber_iff_not_colorable]
  intro hc
```
So I don't now if it is worth it or not, what do you think? Maybe you now a better way to do all those rewrites